src/main/java/com/ays/backend/user/controller/UserController.java

```diff
@@ -0,0 +1,36 @@
+package com.ays.backend.user.controller;
+
+import com.ays.backend.user.controller.payload.request.SignUpRequest;
+import com.ays.backend.user.controller.payload.response.SignUpResponse;
+import com.ays.backend.user.exception.UserAlreadyExistsException;
+import com.ays.backend.user.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * User controller to perform user related api operations.
+ */
+@RestController
+@RequestMapping("/api/v1/user")
+@RequiredArgsConstructor
+public class UserController {
+    private final UserService userService;
+
+    @PostMapping
+    public ResponseEntity<SignUpResponse> registerUser(@RequestBody @Validated SignUpRequest signUpRequest) {
```

Comment:
Use `@Valid` annotation instead `@Validated` for model attributes

**Source :** [Differences in @Valid and @Validated Annotations in Spring](https://www.baeldung.com/spring-valid-vs-validated)